frontend/src/store/teachers.js

```diff
@@ -87,4 +87,49 @@ export const teacherGetters = {
   teachers(state) {
     return state.teachers;
   },
+
+  suggestedTeachers(state) {
+    return (studentId) => {
+
+      // 1. Find the student using the ID
+      const student = state.students.find(student => student.StudentID === studentId)
+      if (!student) {
+        return []
+      }
+      // 2. Find their native language
+      const studentNativeLanguage = student.NativeLanguageID
+      
+      // 3. Get unmatched teachers whose first or second language equals student's native language
+      const relevantTeachers = state.teachers.filter(teacher => {
+        const doesNativeLanguageMatch = teacher.NativeLanguageID === studentNativeLanguage
+        const doesSecondLanguageMatch = teacher.SecondLanguageID === studentNativeLanguage
+
+        return ((doesNativeLanguageMatch || doesSecondLanguageMatch))
+      })
+
+      // 4. Sort teachers by date joined AND status
+      relevantTeachers.sort((t1, t2) => {
+        if (t1.status.Description === "UNMATCHED" && t2.status.Description === "MATCHED") {
+          return -1
+        } 
+        if (t2.status.Description === "UNMATCHED" && t1.status.Description === "MATCHED") {
+          return 1
```

Comment:
This is not triggered because teachers are initialized to the `Screening` status instead of the `Unmatched` status. Issue #105 has been created for this because teachers should be initialized to `Unmatched` instead.